package/MDAnalysis/coordinates/TRR.py

```diff
@@ -13,111 +13,114 @@
-   ts = MDAnalysis.coordinates.TRR.Timestep(N)  # TRR time step
-                                                # N of atoms is passed.
-   for frame,mode in enumerate(modes[4:16]):
-       ts.lmbda = -1
-
-       ts.frame = frame         # manually change the frame number
-       ts._frame = frame - 1
-
-       if frame<=1:
-          ts.positions = xav
-       else:
-          ts.positions = mode.scaledToNorm(1.).array*10   # nm to angstroms
-       if frame <= 1:
-          ts.time = frame-1
-       else:
-          ts.time = mode.frequency
-       W.write(ts)             # converts angstrom to nm for gmx
-
-    W.close()
-
-.. _MMTK: http://dirac.cnrs-orleans.fr/Manuals/MMTK/index.html
-
-.. _`recipe by Ramon Crehuet`: https://github.com/MDAnalysis/mdanalysis/issues/79
-
-Module reference
-----------------
-
-.. autoclass:: Timestep
-   :members:
-   :inherited-members:
-
-.. autoclass:: TRRReader
-   :members:
-   :inherited-members:
-
-.. autoclass:: TRRWriter
-   :members:
-   :inherited-members:
-"""
-
-from .xdrfile.TRR import TRRReader, TRRWriter, Timestep
+from .XDR import XDRBaseReader, XDRBaseWriter
+from ..lib.formats.xdrlib import TRRFile
+from ..lib.mdamath import triclinic_vectors
+
+
+class TRRWriter(XDRBaseWriter):
+    format = 'TRR'
+    units = {'time': 'ps', 'length': 'nm', 'velocity': 'nm/ps',
+             'force': 'kJ/(mol*nm)'}
+    _file = TRRFile
+
+    def write_next_timestep(self, ts):
+        """Write timestep object into trajectory.
+
+        Parameters
+        ----------
+        ts: TimeStep
+
+        See Also
+        --------
+        <FormatWriter>.write(AtomGroup/Universe/TimeStep)
+        The normal write() method takes a more general input
+        """
+        xyz = None
+        if ts.has_positions:
+            xyz = ts.positions.copy()
+            if self._convert_units:
+                self.convert_pos_to_native(xyz)
+
+        velo = None
+        if ts.has_velocities:
+            velo = ts.velocities.copy()
+            if self._convert_units:
+                self.convert_velocities_to_native(velo)
+
+        forces = None
+        if ts.has_forces:
+            forces = ts.forces.copy()
+            if self._convert_units:
+                self.convert_forces_to_native(forces)
+
+        time = ts.time
+        step = ts.frame
+
+        if self._convert_units:
+            dimensions = self.convert_dimensions_to_unitcell(ts, inplace=False)
+
+        box = triclinic_vectors(dimensions)
+
+        self._xdr.write(xyz, velo, forces, box, step, time, 1, self.n_atoms)
+
+
+class TRRReader(XDRBaseReader):
+    """The Gromacs TRR trajectory format is a lossless format. The TRR format can
+    store *velocoties* and *forces* in addition to the coordinates. It is also
+    used by other Gromacs tools to store and process other data such as modes
+    from a principal component analysis.
+
+    Parameter
+    ---------
+    filename: str
```

Comment:
With the docstrings, I don't know if the colon has to have spaces either side (`filename : str`)

In this there seems to be spaces?
http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_numpy.html